src/arviz_stats/summary.py

```diff
@@ -215,3 +216,129 @@ def summary(
+        If dict, the keys are the variable names and the values are tuples with the lower
+        and upper bounds of the ROPE. The keys must be in `var_names`.
+    ci_prob : float, optional
+        Probability for the credible interval. Defaults to ``rcParams["stats.ci_prob"]``.
+    ci_kind : {"hdi", "eti"}, optional
+        Type of credible interval. Defaults to ``rcParams["stats.ci_kind"]``.
+        If `kind` is stats_median or all_median, `ci_kind` is forced to "eti".
+
+    Returns
+    -------
+    xarray.Dataset
+
+    See Also
+    --------
+    arviz.summary : Compute summary statistics and or diagnostics.
+    arviz.hdi : Compute highest density interval (HDI).
+    arviz.eti : Compute equal tail interval (ETI).
+
+    Examples
+    --------
+    Apply the same ROPE to a subset of variables:
+
+    .. ipython::
+        In [1]: from arviz_base import load_arviz_data
+           ...: from arviz_stats import ci_in_rope
+           ...: data = load_arviz_data("centered_eight")
+           ...: ci_in_rope(data, var_names=["mu", "tau"], rope=(-0.5, 0.5))
+
+    Apply different ROPEs to each variable:
+
+    .. ipython::
+        In [1]: ci_in_rope(data, rope={"mu": (-0.5, 0.5), "tau": (0.1, 0.2), "theta": (-0.1, 0.1)})
+
+    References
+    ----------
+    .. [1] Kruschke. Doing Bayesian Data Analysis, Second Edition: A Tutorial with R,
+        JAGS, and Stan. Academic Press, 2014. ISBN 978-0-12-405888-0.
+    """
+    if ci_kind not in ("hdi", "eti", None):
+        raise ValueError("ci_kind must be either 'hdi' or 'eti'")
+    if ci_prob is None:
+        ci_prob = rcParams["stats.ci_prob"]
+    if ci_kind is None:
+        ci_kind = rcParams["stats.ci_kind"] if "stats.ci_kind" in rcParams else "eti"
+    if sample_dims is None:
+        sample_dims = rcParams["data.sample_dims"]
+
+    dataset = extract(
+        data,
+        var_names=var_names,
+        filter_vars=filter_vars,
+        group=group,
+        combined=False,
+        keep_dataset=True,
+    )
+
+    if ci_kind == "eti":
+        c_i = dataset.azstats.eti(prob=ci_prob, dims=sample_dims).rename({"quantile": "ci_kind"})
+    else:
+        c_i = dataset.azstats.hdi(prob=ci_prob, dims=sample_dims).rename({"hdi": "ci_kind"})
+
+    stacked = dataset.stack(sample=sample_dims)
+
+    ci_low = c_i.sel(ci_kind="lower")
+    ci_high = c_i.sel(ci_kind="higher")
+
+    in_ci = (stacked >= ci_low) & (stacked <= ci_high)
+
+    if isinstance(rope, tuple):
+        rope_low = xr.full_like(ci_low, rope[0])
+        rope_high = xr.full_like(ci_high, rope[1])
+    elif isinstance(rope, dict):
+        rope_low = xr.full_like(ci_low, fill_value=np.nan)
+        rope_high = xr.full_like(ci_high, fill_value=np.nan)
+
+        for var, (low, high) in rope.items():
+            if var in rope_low:
+                rope_low[var] = low
+                rope_high[var] = high
+    else:
+        raise ValueError("`rope` must be a tuple or dict")
```

Comment:
I think this could be handled by `references_to_dataset` with maybe an extra check to make sure all variables in input data are present in rope dataset?